Adding os_subnets_facts module

dagnello opened the pull request:
There can be instances during an Ansible play where the list of subnets
currently available from OpenStack is required.  This update provides
subnet list functionality as a new os_subnets_facts module.

bcoca commented:
cc @emonty @Shrews @j2sol

omgjlk commented:
Lets have the listing exist in only one module, rather than in two places. So lets keep the facts module and not modify the existing `os_subnet` module.

dagnello commented:
@j2sol yes, I would like to not include the os_subnet commit.

dagnello commented:
@bcoca @emonty @Shrews @j2sol ready

dagnello committed:
Adding new os_subnets_facts module

There can be instances during an Ansible play where the list of subnets
currently available from OpenStack is required.  This update provides
subnet list functionality as a new os_subnets_facts module.

Shrews commented:
LGTM  :+1:

Shrews commented:
:shipit:

omgjlk commented on cloud/openstack/os_subnets_facts.py:
```diff
@@ -0,0 +1,152 @@
+    name:  subnet1
+- debug: var=openstack_subnets
+
+# Gather facts about a previously created subnet with filter (not name and
```
s/not/note/

omgjlk commented:
:+1:

dagnello commented:
@gregdek one more :)

dagnello commented:
cc @robynbergeron

dagnello commented:
cc @bcoca patch is ready
